lib/SIL/IR/SILPrinter.cpp

```diff
@@ -3215,10 +3215,13 @@ static StringRef getLinkageString(SILLinkage linkage) {
   switch (linkage) {
   case SILLinkage::Public: return "public ";
   case SILLinkage::PublicNonABI: return "non_abi ";
+  case SILLinkage::Package: return "package ";
+  case SILLinkage::PackageNonABI: return "non_abi "; // Q: return non_abi for package nonabi?
```

Comment:
is it safe to return the same value as L3217 here? @eeckstein @slavapestov

Reply:
The textual SIL should represent the internal SIL data structure. Printing `non_abi` for `PackageNonABI` would hide this information in textual SIL. So you should return `package_non_abi`. And also handle this in the SILParser.